Zynq hybrid: Microblaze BSP build fails due to missing openmac_cfg.h

Building the Microblaze BSP for the Zynq hybrid design fails due to missing openmac_cfg.h file.

Log:

```
[ 22%] Generating bsppcp/pcp/lib/libxil.a
Starting SDK. This could take few seconds... done
INFO: [Hsi 55-1698] elapsed time for repository loading 0 seconds
CRITICAL WARNING: [Hsi 55-1608] VERSION option is not used in file /somewhere/hardware/ipcore/xilinx/components/pcores/axi_openmac_v1_02_a/data/axi_openmac_v2_1_0.mdd. Use 'OPTION VERSION = <>;' && 'OPTION NAME = <>;' in MDD.
CRITICAL WARNING: [Hsi 55-1608] VERSION option is not used in file /somewhere/hardware/ipcore/xilinx/components/pcores/axi_hostinterface_v1_02_a/data/axi_hostinterface_v2_1_0.mdd. Use 'OPTION VERSION = <>;' && 'OPTION NAME = <>;' in MDD.
CRITICAL WARNING: [Hsi 55-1608] VERSION option is not used in file /somewhere/hardware/ipcore/xilinx/components/pcores/axi_openmac_v1_02_a/data/axi_openmac_v2_1_0.mdd. Use 'OPTION VERSION = <>;' && 'OPTION NAME = <>;' in MDD.
CRITICAL WARNING: [Hsi 55-1608] VERSION option is not used in file /somewhere/hardware/ipcore/xilinx/components/pcores/axi_hostinterface_v1_02_a/data/axi_hostinterface_v2_1_0.mdd. Use 'OPTION VERSION = <>;' && 'OPTION NAME = <>;' in MDD.
Hardware platform project 'hw_platform' created successfully.
/opt/toolchains/xilinx/SDK/2016.2/gnu/microblaze/lin
CRITICAL WARNING: [Hsi 55-1564] TCL file /somewhere/hardware/ipcore/xilinx/components/pcores/axi_openmac_v1_02_a/data/openmac.tcl doesn't exist for driver openmac
CRITICAL WARNING: [Hsi 55-1564] TCL file /somewhere/hardware/ipcore/xilinx/components/pcores/axi_openmac_v1_02_a/data/openmac.tcl doesn't exist for driver openmac
BSP project 'bsppcp' created successfully.
Running Make include in pcp/libsrc/bram_v4_1/src
Running Make include in pcp/libsrc/cpu_v2_5/src
Running Make include in pcp/libsrc/ddrps_v1_0/src
Running Make include in pcp/libsrc/gpio_v4_1/src
Running Make include in pcp/libsrc/intc_v3_5/src
Running Make include in pcp/libsrc/standalone_v5_5/src
Running Make include in pcp/libsrc/uartlite_v3_2/src
Running Make include in pcp/libsrc/uartps_v3_1/src
Running Make libs in pcp/libsrc/bram_v4_1/src
Compiling bram
mb-ar: creating ../../../lib/libxil.a
Running Make libs in pcp/libsrc/cpu_v2_5/src
Compiling cpu
Running Make libs in pcp/libsrc/ddrps_v1_0/src
Compiling ddrps
Running Make libs in pcp/libsrc/gpio_v4_1/src
Compiling gpio
Running Make libs in pcp/libsrc/intc_v3_5/src
Compiling intc
Running Make libs in pcp/libsrc/standalone_v5_5/src
Compiling standalone
Running Make libs in pcp/libsrc/uartlite_v3_2/src
Compiling uartlite
Running Make libs in pcp/libsrc/uartps_v3_1/src
Compiling uartps
Finished building libraries
...
In file included from /somewhere/stack/src/kernel/edrv/edrv-openmac.c:49:0:
/somewhere/stack/include/target/openmac.h:45:25: fatal error: openmac_cfg.h: No such file or directory
compilation terminated.
gmake[2]: *** [proj/generic/liboplkmndrv-dualprocshm/CMakeFiles/oplkmndrv-dualprocshm.dir/__/__/__/src/kernel/edrv/edrv-openmac.c.obj] Error 1
gmake[1]: *** [proj/generic/liboplkmndrv-dualprocshm/CMakeFiles/oplkmndrv-dualprocshm.dir/all] Error 2
gmake: *** [all] Error 2
```

Workaround:
Vivado selects the ISE compatible ipcore of openMAC, thus, removing it before building works as a temporary workaround.

```
rm -rf hardware/ipcore/xilinx/components/pcores
```

--- a/hardware/cmake/xilinx/genbsp.cmake
+++ b/hardware/cmake/xilinx/genbsp.cmake
@@ -2,7 +2,7 @@
 #
 # CMake macro for generating the board support package for Vivado Microblaze
 #
-# Copyright (c) 2016, Kalycito Infotech Private Limited
+# Copyright (c) 2017, Kalycito Infotech Private Limited
 # All rights reserved.
 #
 # Redistribution and use in source and binary forms, with or without
@@ -49,7 +49,7 @@ MACRO(GENERATE_BSP EXAMPLE_NAME XIL_DEMO_DIR XIL_BSP_TARGET_DIR PROCESSOR_NAME T
         ADD_CUSTOM_COMMAND(
             OUTPUT ${XIL_BSP_TARGET_DIR}/${PROCESSOR_NAME}/lib/libxil.a
             DEPENDS ${VIVADO_DEMO_DIR}/system/system.sdk/${BITS_SYSTEM_NAME}.hdf
-            COMMAND xsdk -batch -source ${ARCH_TOOLS_DIR}/genbspvivado.tcl ${XIL_BSP_TARGET_DIR}/.. ${XIL_DEMO_DIR}/../../../ipcore/xilinx ${BSP_VIVADO_DIR}/system/system.sdk/${BITS_SYSTEM_NAME}.hdf ${PROCESSOR_NAME}
+            COMMAND xsdk -batch -source ${ARCH_TOOLS_DIR}/genbspvivado.tcl ${XIL_BSP_TARGET_DIR} ${XIL_PLK_IPCORE_REPO}/components/drivers ${BSP_VIVADO_DIR}/system/system.sdk/${BITS_SYSTEM_NAME}.hdf ${PROCESSOR_NAME}
             COMMAND make -C ${XIL_BSP_TARGET_DIR}
             WORKING_DIRECTORY ${XIL_BSP_TARGET_DIR}/..
         )
@@ -61,7 +61,14 @@ MACRO(GENERATE_BSP EXAMPLE_NAME XIL_DEMO_DIR XIL_BSP_TARGET_DIR PROCESSOR_NAME T
         )
 
         # Add all generated files to clean target
-        SET(ADD_CLEAN_FILES ${ADD_CLEAN_FILES} ${XIL_BSP_TARGET_DIR})
+        SET(ADD_CLEAN_FILES ${ADD_CLEAN_FILES}
+                            ${XIL_BSP_TARGET_DIR}/../hw_platform
+                            ${XIL_BSP_TARGET_DIR}/../SDK.log
+                            ${XIL_BSP_TARGET_DIR}/../.metadata
+                            ${XIL_BSP_TARGET_DIR}/../.xil
+                            ${XIL_BSP_TARGET_DIR}/../system.mss
+                            ${XIL_BSP_TARGET_DIR}
+           )
     ELSE()
         MESSAGE(FATAL_ERROR "libgen was not found in system PATH or ISE installation")
     ENDIF()

--- a/tools/xilinx-microblaze/genbspvivado.tcl
+++ b/tools/xilinx-microblaze/genbspvivado.tcl
@@ -2,7 +2,7 @@
 #
 # Generate the bsp for Zynq Vivado design
 #
-# Copyright (c) 2016, Kalycito Infotech Private Limited
+# Copyright (c) 2017, Kalycito Infotech Private Limited
 # All rights reserved.
 #
 # Redistribution and use in source and binary forms, with or without
@@ -36,13 +36,18 @@ if { $argc != 4 } {
         #puts [expr [lindex $argv 0] + [lindex $argv 1]]
     }
 
-#Local parameters
+# Local parameters
 set hw_project hw_platform
 set platform standalone
 
+# To avoid error during recompilation of bsp
+file delete -force [lindex $argv 0]/../.metadata
+file delete -force [lindex $argv 0]/../.xil
+file delete -force [lindex $argv 0]
+
 # Create workspace
 # Already set from CMake
-setws [lindex $argv 0]
+setws [lindex $argv 0]/..
 
 # Set local repo
 repo -set [lindex $argv 1]

